manifests/init.pp

```diff
@@ -221,6 +232,24 @@
     restart    => 'PATH=/usr/bin:/bin systemctl reload nftables',
   }
 
+  if $purge_unmanaged_rules {
+    # Reload the nftables ruleset from the on-disk ruleset if there are differences or it is absent. -s must be used to ignore counters
+    exec { 'nftables_memory_state_check':
+      command => 'echo "reloading nftables"',
+      path    => ['/usr/sbin', '/sbin', '/usr/bin', '/bin'],
+      unless  => "/usr/bin/test -s ${inmem_rules_hash_file} -a \"$(nft -s list ruleset | sha1sum)\" = \"$(cat ${inmem_rules_hash_file})\"",
```

Comment:
can you please update the command and unless parameter to be arrays and not strings? That's a new best practice. Perforce already has a puppet-lint plugin for it and we will probably roll that out in the future as well.

Reply:
this was tricky with the length of the unless.  I have put it in an array but all as 1 element, it failed with splitting it up in my testing.  The second exec to generate the has I couldn't get to work in an array at all, even as 1 element.  Any suggestions?

Reply:
Thanks for the tip to add shell provider, that has allowed the second exec to be in an array.  I tried breaking the execs into elements, but couldn;t get them to work in various permutations.  I think they will need to remain as single elements.